### DISCUSSION TITLE
Issue with Vercel subdomain - cname

### DISCUSSION THREAD

**Discussion by Jerga99** [vercel/community#34] [Help] [answered]
Hello Guys!

I will try to describe my problem as best as I can (:

I have sub-domain assigned by Teachable -> [eincode.teachable.com](https://eincode.teachable.com)
I have domain on GoDaddy -> [eincode.com](https://eincode.com) but I have setup Vercel NameServers(NS) for [eincode.com](https://eincode.com)

I would like to create subdomain [academy.eincode.com](https://academy.eincode.com) then I would like to use sub-domain [academy.eincode.com](https://academy.eincode.com) to point to my teachable school [eincode.teachable.com ](https://eincode.teachable.com)

I followed steps explained on Teachable (basically just to set up CNAME),  https://support.teachable.com/hc/en-us/articles/115000449711-Custom-Subdomains

Since my NS are assigned to Vercel I had to set it up on Vercel DNS. Here is my CNAME on Vercel for [eincode.com](https://eincode.com) ->
https://i.imgur.com/7oCtKwB.png

After all, I am getting SSL related issues. you can see when you navigate to https://academy.eincode.com/
Am I doing something wrong ? What can be an issue ?

I also tried to do same setup with my other GoDaddy domain [filipjerga.com](https://filipjerga.com) which has GoDaddy NS. I entered cname -> 
https://i.imgur.com/f9oLwFP.png

Everything is working as expected when visiting [test.filipjerga.com](https://test.filipjerga.com)

Why it's working for goDaddy but not for Vercel ?

I will be super thankful for any help/advice or a tip

Take care,
Filip

**Comment by MartinKnopf** [comment, marked as answer]
Looks like you figured it out by now. For anyone coming here via google:

**TL;DR**
In order to forward an external subdomain to a Vercel project you have to:
1. Add a `CNAME` record for your subdomain pointing to `cname.vercel-dns.com` (e.g. `mysubdomain -> cname.vercel-dns.com`)
2. Add your subdomain to the Vercel project using the [UI](https://vercel.com/docs/custom-domains) or the [API](https://vercel.com/docs/api#endpoints/domains/add-a-new-domain) or the [CLI](https://vercel.com/docs/cli#commands/domains)

Vercel will automatically generate the certificate so that `https://mysubdomain.example.com` will work.

**Reply by codearemo** [reply]
Please I ran this:
vercel dns add [domain] [subdomain] [A || AAAA || ALIAS || CNAME || TXT] [value]

But I still get the SSH error....please what am I doing wrong

**Reply by DougAnderson444** [reply]
You need to replace the stuff in [brackets] with your own values, and choose from only one of the values separated by ||

**Reply by bvschwartz** [reply]
How does it know which Vercel project to run?

**Comment by znwhite** [comment]
I am having an issue getting a subdomain to work on Vercel project through Namecheap. I have one project as example.com and it is working, then a second project that I am trying to point to app.example.com using a CNAME record in Namecheap, but the subdomain does not work.

My Namecheap setup currently:

**example.com**
A Record - @ - 76.76.21.21
CNAME Record  - www - cname.vercel-dns.com
CNAME Record - app - cname.vercel-dns.com

**Reply by amyegan** [reply]
Hi @znwhite! Can you tell me more about how it is not working (i.e. SSL cert not generating, 404, etc.)?

**Reply by rajavnr2003** [reply]
hi, please let us know if you have resolved this...having same issue.

update:

after updating CNAME records, give some time it will work.

**Reply by znwhite** [reply]
Hi all, can confirm what @rajavnr2003 just said. I had to wait and it worked. The real issue for me was that my wifi router wasn't clearing its cache though. So I had to turn my wifi router on and off and it worked (I noticed that whenever I was off of my home's wifi, it was working...)

**Comment by nickdelja** [comment]
I am having an issue where the cert is stuck in a generating state:
<img width="734" alt="image" src="https://user-images.githubusercontent.com/6005805/195454927-d8429dfa-dae4-4280-83cf-dd32a18b71bf.png">

**Comment by pariweshh** [comment]
Hi @Jerga99 it seems like you have fixed the issue, would you mind sharing how?  I am facing the same issue. Thanks in advance.
